packages/app-js/src/index.tsx

```diff
@@ -158,12 +217,40 @@ class App extends React.PureComponent<Props, State> {
   }
 
   private selectExample = (value: string) => {
-    const snippet = snippets.find(obj => obj.value === value);
+    if (value.length) {
+      const { options } = this.state;
+      const option = options.find(obj => obj.value === value);
 
-    localStorage.setItem('app-js-snippet', value);
-    this.setState({ code: (snippet ? snippet.code : ''), snippet: value });
+      localStorage.setItem('polkadot-app-js-selected', value);
+      this.setState({ code: (option ? option.code : ''), snippet: value });
+    }
+  }
+
+  private saveSnippet = (): void => {
+    const { code, customExamples, snippetName } = this.state;
+
+    const snapshot: Snippet = {
+      ...customExample,
+      code,
+      text: snippetName,
+      value: `custom-${Date.now()}`
+    };
+
+    localStorage.setItem('polkadot-app-js-examples', JSON.stringify([snapshot, ...customExamples]));
```

Comment:
May make sense to have these strings (key names) as constants, in the file, i.e.

```
const STORE_EXAMPLES = 'polkadot-app-js-examples';
```

(just protects agains possible typos)

Reply:
Good idea, I moved it.